presto-main/src/test/java/io/prestosql/sql/planner/optimizations/TestAddExchangesPlans.java

```diff
@@ -162,6 +169,27 @@ public void testNonSpillableBroadcastJoinAboveTableScan()
                                                         tableScan("region", ImmutableMap.of("regionkey", "regionkey"))))))));
     }
 
+    @Test
+    public void testForcePartitioningBelowMarkDistinctNode()
+    {
+        assertDistributedPlan(
+                "SELECT count(orderkey), count(distinct orderkey), custkey , count(1) FROM ( SELECT * FROM (VALUES (1, 2)) as t(custkey, orderkey) UNION ALL SELECT 3, 4) GROUP BY 3",
+                Session.builder(getQueryRunner().getDefaultSession())
+                        .setSystemProperty(ENABLE_FORCED_PARTITIONING_BELOW_MARK_DISTINCT, "true")
```

Comment:
test the same query when this session property is set to `false`

Also we need correctness tests for this property (run this and similar queries and validate results in distributed query runner).